src/pages/resultsView/cnSegments/CNSegments.tsx

```diff
@@ -46,32 +48,32 @@ class CNASegmentIframe extends React.Component<CNSegmentsIframeProps,{}>{
         
     }
 
-    downloadSegmentData(){
+    downloadSegmentData() {
             const self = this;
-            var xhr = new XMLHttpRequest(),
-                sendData = `cancerStudyId=${self.props.studyId}&sampleIds=${self.props.sampleIds.join(',')}`;
+            const xhr = new XMLHttpRequest();
+            const sendData = `cancerStudyId=${self.props.studyId}&sampleIds=${self.props.sampleIds.join(',')}`;
+
             xhr.onreadystatechange = function() {
-                var a:any;
                 if (xhr.readyState === 4 && xhr.status === 200) {
-                    // Making a downloadable link
-                    a = document.createElement('a');
-                    a.href = window.URL.createObjectURL(xhr.response);
-                    a.download = self.props.studyId + '_segments.seg';
-                    a.style.display = 'none';
-                    document.body.appendChild(a);
-                    //triger download
-                    a.click();
-                    setTimeout(function(){
-                       $(a).remove();
-                    },200);
+                    const reader = new FileReader();
+
+                    reader.addEventListener('loadend', (e: any) => {
+                        // seems like the downloaded string is wrapped with quotes,
+                        // also it contains "\\n", "\\t", instead of "\n", and "\t"
```

Comment:
where is the data coming from?  i'm just curious why it's escaped like this.   would be good to include that in comment.   if we don't control it, then indicate that.

Reply:
Seems like it is coming from `api-legacy/segmentfile` endpoint, but I haven't really checked the actual code. It might be possible to fix the problem there.

Reply:
Here is the legacy API code: https://github.com/cBioPortal/cbioportal/blob/94273890828889b775ef0b7d698ab617c72edf0a/business/src/main/java/org/mskcc/cbio/portal/service/CNSegmentServiceImpl.java#L29

Not sure why we end up with a wrapped and escaped string in the end.